flood_adapt/object_model/hazard/forcing/netcdf.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+
+@staticmethod
+def validate_netcdf_forcing(
+    ds: xr.Dataset, required_vars: set[str], required_coords: set[str]
+) -> xr.Dataset:
+    """Validate a forcing dataset by checking for required variables and coordinates."""
+    if not required_vars.issubset(ds.data_vars):
+        missing_vars = required_vars - set(ds.data_vars)
+        raise ValueError(
+            f"Missing required variables for netcdf forcing: {missing_vars}"
+        )
```

Comment:
Should we also check that the required vars are not empty?

Reply:
and for that matter, if we always use metric internally, do we need to check that it is in metric?

Reply:
I think sfincs also requires a particular ordering in the spacial dimensions, so we should check that too. iirc it was `[y,x]` but that I'm not totally sure about